iroha/src/sumeragi.rs

```diff
@@ -116,6 +120,11 @@ impl Sumeragi {
                     self.invalidated_blocks_hashes.clone(),
                 )
                 .validate(&*wsv.read().await);
+            self.events_sender
+                .send(crate::event::Occurrence::Created(
+                    crate::event::Entity::Block(block.clone()),
+                ))
+                .await;
```

Comment:
Block is created on a leader, so if the client wants all block created events it will have to connect to a different peer every time a block is committed or a view change happens.

Reply:
And how the client will know what peer is a leader?

Reply:
Maybe we can propagate these events over the network, so every peer is aware of them.

Reply:
The test `client_subscribe_to_changes_request_should_receive_changes` is passing right now because we have only one peer there.

Reply:
I will work on events submissions in the next PR.